group incidents by year

hansmannj opened the pull request:
Suggestion:
How about grouping incidents by year and only expanding the current year by default, to improve readability?
Esp. useful in years with many incidents

[Test link](https://sys-docs.dev.bgdi.ch/preview/feat_group_incidents_by_year/index.html)

hansmannj committed:
group incidents by year

now grouping incidents by year and only expanding the current
year by default, to improve readability.
Esp. useful in years with many incidents

hansmannj committed:
Nested grouping of details

- Group incidents by year
- Group incidents themselves, only keeping the most current / recent one expanded by default
- Added a section on how to add a new incident to the status page in the README.md

asteiner-swisstopo commented:
I like the idea of grouping by year.

How about collapsing _all_ years by default?
<img width="2147" height="946" alt="image" src="https://github.com/user-attachments/assets/07b8c9af-7eb2-484b-963e-d9e77a11a12f" />

This way, the reader sees immediately only the most relevant section: The current status. Also, the instructions on how to archive incidents would get simpler.

hansmannj committed:
Adding third level sections

Now we have three levels of containers. One that starts with 5 colons for the years,
another one starting with 4 colons for the incidents and a last one, starting with 3 colons
for the potentially multipe messages inside the same incident.
Also, everything is collapsed by default

hansmannj committed:
expand all incidents by default

asteiner-swisstopo reviewed (changes requested):
I'm not a fan because of the additional complexity but I agree that it looks tidier than before, so fine if you want to keep that.

What I would try to improve for sure is the README, see the comment below.
